File: rustsem-java/cl_tests/tests/t2.m

#allocateInit(2,i32,valcl(1),valcl(2))
#allocateInit(4,i32,valcl(1),valcl(2),valcl(4),#shr(1,2,#sl(1)))
#allocateInit(3,i32,valcl(1),valcl(2),valcl(3))


#allocStack

#allocStack

#allocStack

#write(#sl(0),#own(#blk(0)))
#write(#sl(1),#Shr(#sl(0)))
#write(#sl(2),#Mut(#sl(0)))

#write(#dRef(#sl(1)),#Shr(#sl(0)))

File: rustsem-java/rust_tests/rv/rduninit/t2.rs

struct A{
       a:i32
}

struct B{
     b1:i32,
     b2:A
}

fn main() {
   let a = A{a:1};
   let b = B{b1:2,b2:a};

   let c = b.b2;

   let d = b.b2;
}